src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2500,6 +2500,9 @@ std::string Creature::GetScriptName() const
 
 uint32 Creature::GetScriptId() const
 {
+    if (CreatureData const* creatureData = GetCreatureData())
```

Comment:
My point is, if there is no script defined in `creature|gameobject`, it won't fallback to the one in `(creature|gameobject)_template`.